### PR TITLE
Aws eks kubeconfig, test client and snapshot id logging fixes

### PR DESCRIPTION
**Pull request by sid255**
# Description

Not fix but misc changes to : 
- [ ] add token to aws generate kubeconfig
- [ ] add DeleteAllClustersInRegion method to test client
- [ ] add base64 encoded token to test client
- [ ] add snapshot id log

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

# why

Easier access to clusters using spawner

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: cmd/testclient/client.go [hunk 1]
```diff
@@ -276,6 +279,36 @@ func main() {
...
+			sugar.Errorw("error getting clusters", "account", getClustersReq.AccountName, "provider", getClustersReq.Provider, "region", getClustersReq.Region, "error", err)
+			os.Exit(1)
+		}
+		clusters := make([]string, 0)
+		for _, cluster := range v.Clusters {
+			clusters = append(clusters, cluster.Name)
+		}
+		logger.Sugar().Infow("deleting following clusters", "clusters", clusters)
+
+		for _, cluster := range clusters {
```
Comment (NishanthSpShetty) [inline cmd/testclient/client.go:338]: can we use v.Clusters directly here ? the above looping isnt required i think

File: cmd/testclient/client.go [hunk 2]
```diff
@@ -276,6 +279,36 @@ func main() {
...
+				Provider:    provider,
+				Region:      region,
+				AccountName: accountName,
+				ClusterName: cluster,
+				ForceDelete: true,
+			}
+			v, err := client.DeleteCluster(context.Background(), req)
+			if err != nil && err.Error() != "" {
+				sugar.Errorw("error deleting cluster", "cluster", cluster, "error", err)
+				os.Exit(1)
```
Comment (NishanthSpShetty) [inline cmd/testclient/client.go:350]: nitpick: avoid using os.Exit unless its absolutely necessory. in large application, there are paths which rely on defer calls, Exit wont allow that to execute.

File: pkg/service/aws/kubec.go [hunk 3]
```diff
@@ -50,15 +50,7 @@ func (ctrl AWSController) GetKubeConfig(ctx context.Context, req *proto.GetKubeC
...
-		Exec: &clientcmdapi.ExecConfig{
-			Command: "aws",
-			Args: []string{
-				"--region", region,
-				"eks", "get-token",
-				"--cluster-name", clusterName,
-			},
-			APIVersion: "client.authentication.k8s.io/v1alpha1",
-		},
+		Token: kubeConfig.BearerToken,
```
Comment (NishanthSpShetty) [inline pkg/service/aws/kubec.go:53]: this will change the bahaviour of the kubeconfig file generated using spawner cli too, do we want that happen?

Comment (sid255) [reply pkg/service/aws/kubec.go:53]: Yup

File: cmd/testclient/client.go (hunk 2 above)
Comment (sid255) [reply cmd/testclient/client.go:350]: This will only exit the testclient cli app and not spawner application

File: cmd/testclient/client.go (hunk 1 above)
Comment (sid255) [reply cmd/testclient/client.go:338]: The above looping is just to log all the clusters we are deleting for user to see and terminate if something doesn't look good. Ideally should be taking a stdin input to confirm but we are not for now.

File: pkg/service/aws/kubec.go (hunk 3 above)
Comment (NishanthSpShetty) [reply pkg/service/aws/kubec.go:53]: token is very short lived, 15min

File: cmd/testclient/client.go (hunk 2 above)
Comment (NishanthSpShetty) [reply cmd/testclient/client.go:350]: yes, Its just a suggestion. here return also would work the same.

File: cmd/testclient/client.go (hunk 1 above)
Comment (NishanthSpShetty) [reply cmd/testclient/client.go:338]: we can log `v.Clusters` right ?

Comment (sid255) [reply cmd/testclient/client.go:338]: v.Clusters is a huge json and difficult to read.

File: cmd/testclient/client.go (hunk 2 above)
Comment (sid255) [reply cmd/testclient/client.go:350]: Removed it altogether so we can try and delete other clusters.

File: pkg/service/aws/kubec.go (hunk 3 above)
Comment (sid255) [reply pkg/service/aws/kubec.go:53]: True, that could be an issue. How about we pass a flat in `GetKubeConfig` request to get raw token?

Comment (NishanthSpShetty) [reply pkg/service/aws/kubec.go:53]: yeah, i think that would helpful. even if its 15min its very useful for our testing, adding flag for command would help use to allow testing different cluster.

File: cmd/testclient/client.go [hunk 4]
```diff
@@ -320,6 +323,35 @@ func main() {
...
+			req := &proto.ClusterDeleteRequest{
+				Provider:    provider,
+				Region:      region,
+				AccountName: accountName,
+				ClusterName: cluster,
+				ForceDelete: true,
+			}
+			v, err := client.DeleteCluster(context.Background(), req)
+			if err != nil && err.Error() != "" {
+				sugar.Errorw("error deleting cluster", "cluster", cluster, "error", err)
```
Comment (NishanthSpShetty) [inline cmd/testclient/client.go:349]: may be add continue when fails.

lets simplify this `if err != nil && err.Error() != ""` to `if err != nil`.

Comment (sid255) [reply cmd/testclient/client.go:349]: fixed

File: proto/netbookai/spawner/spawner.proto [hunk 5]
```diff
@@ -425,6 +425,7 @@ message GetKubeConfigRequest {
   string region = 2;
   string accountName = 3;
   string clusterName = 4;
+  bool getRawToken = 5;
```
Comment (NishanthSpShetty) [inline proto/netbookai/spawner/spawner.proto:428]: minor nitpick: can we remove `get` from field name?

**Pull request merged**